tests/benchmark/compute/instruction/test_system.py

```diff
@@ -571,41 +572,71 @@ def test_selfdestruct_existing(
     )
     factory_caller_address = pre.deploy_contract(code=factory_caller_code)
 
+    # The deployed code length is two-bytes. The dominant factor
+    # is the static cost, plus a few glue opcodes/memory-expansion costs.
+    estimated_gas_per_creation = gas_costs.G_CREATE + 3_000
+    max_creations_per_tx = int(tx_gas_limit // estimated_gas_per_creation)
+    num_setup_txs = math.ceil(num_contracts / max_creations_per_tx)
+
+    setup_txs = []
     with TestPhaseManager.setup():
-        contracts_deployment_tx = Transaction(
-            to=factory_caller_address,
-            gas_limit=env.gas_limit,
-            data=Hash(num_contracts),
-            sender=pre.fund_eoa(),
-        )
+        for i in range(num_setup_txs):
+            count = min(
+                max_creations_per_tx, num_contracts - i * max_creations_per_tx
+            )
+            setup_txs.append(
+                Transaction(
+                    to=factory_caller_address,
+                    gas_limit=tx_gas_limit,
+                    data=Hash(count),
+                    sender=pre.fund_eoa(),
+                )
+            )
 
     code = (
         # Setup memory for later CREATE2 address generation loop.
         # 0xFF+[Address(20bytes)]+[seed(32bytes)]+[initcode keccak(32bytes)]
         Op.MSTORE(0, factory_address)
         + Op.MSTORE8(32 - 20 - 1, 0xFF)
-        + Op.MSTORE(32, 0)
+        + Op.MSTORE(32, Op.CALLDATALOAD(0))  # Starting address from calldata
         + Op.MSTORE(64, initcode.keccak256())
         # Main loop
         + While(
             body=Op.POP(Op.CALL(address=Op.SHA3(32 - 20 - 1, 85)))
             + Op.MSTORE(32, Op.ADD(Op.MLOAD(32), 1)),
-            # Only loop if we have enough gas to cover another iteration plus
-            # the final storage gas.
-            condition=Op.GT(Op.GAS, final_storage_gas + loop_cost),
+            # Loop while we have enough gas AND within target count
+            condition=Op.AND(
+                Op.GT(Op.GAS, final_storage_gas + loop_cost),
+                Op.LT(
+                    Op.MLOAD(32),
+                    Op.ADD(Op.CALLDATALOAD(0), Op.CALLDATALOAD(32)),
+                ),
+            ),
         )
         + Op.SSTORE(0, 42)  # Done for successful tx execution assertion below.
     )
     assert len(code) <= fork.max_code_size()
 
     # The 0 storage slot is initialize to avoid creation costs in SSTORE above.
     code_addr = pre.deploy_contract(code=code, storage={0: 1})
```

Comment:
Following the previous comment, i update the storage to `{0: 42}` here


```suggestion
    code_addr = pre.deploy_contract(code=code, storage={0: 42})
```

Reply:
I'll keep this as is, since I switched to option 3. idea in my previous comment.